packages/flutter_test/lib/src/all_elements.dart

```diff
@@ -23,15 +25,35 @@ Iterable<Element> collectAllElementsFrom(
   return CachingIterable<Element>(_DepthFirstChildIterator(rootElement, skipOffstage));
 }
 
+/// Provides a recursive, efficient, depth first search of an element tree.
+///
+/// [Element.visitChildren] does not guarnatee order, but does guarnatee stable
+/// order. This iterator also guarantees stable order, but will iterate in a
+/// right to left order, e.g.:
+///
+///       1
+///     /   \
+///    2     3
+///   / \   / \
+///  4   5 6   7
+///
+/// Will iterate in order 1, 3, 7, 6, 2, 5, 4. This avoids unnecessary
+/// allocation or CPU time, and performance is important here because this
```

Comment:
How?

Reply:
avoids extra allocations?

Reply:
The previous method involved creating multiple lists, doing addAlls to combine lists, and reverse iterating lists to keep things in LTR order.

Reply:
SG, make sure to document this